server/repository/users.go

```diff
@@ -1,34 +1,65 @@
 package repository
 
 import (
+	"database/sql"
 	"fmt"
 
 	"github.com/src-d/code-annotation/server/model"
 )
 
-// Users is the User repository
+// Users repository
 type Users struct {
-	users []*model.User
+	db *sql.DB
 }
 
-// Create stores a user into the DB
-func (r *Users) Create(m *model.User) error {
-	for _, u := range r.users {
-		if u.ID == m.ID {
-			return nil
-		}
+// NewUsers returns a new Users repository
+func NewUsers(db *sql.DB) *Users {
+	return &Users{db: db}
+}
+
+// Create stores a User into the DB, and returns that new User
+func (repo *Users) Create(user *model.User) error {
+
+	_, err := repo.db.Exec(
+		"INSERT INTO users (login, username, avatar_url, role) VALUES ($1, $2, $3, $4)",
+		user.Login, user.Username, user.AvatarURL, user.Role)
+
+	if err != nil {
+		return err
 	}
-	r.users = append(r.users, m)
-	return nil
+
+	user, err = repo.Get(user.Login)
```

Comment:
do you need this get to get `id`?
this way it's not going to work. Simplified example of what you are doing: https://play.golang.org/p/dlEmRgTL1uN